### PR TITLE
Update 04-pipefilter.md to reflect current episode structure

### PR DESCRIPTION
**Pull request by themkots**
Rearranged the Objectives of the episode to match the current discussion, in order of appearance.

Need some edits more to emphasize the UNIX filters 'philosophy', i.e. 'programs that do one thing, and do it well'.

_If this pull request addresses an open issue on the repository, please add 'Closes #NN' below, where NN is the issue number._

_Please briefly summarise the changes made in the pull request, and the reason(s) for making these changes._

_If any relevant discussions have taken place elsewhere, please provide links to these._

<details>

For more guidance on how to contribute changes to a Carpentries project, please review [the Contributing Guide](CONTRIBUTING.md) and [Code of Conduct](https://docs.carpentries.org/topic_folders/policies/code-of-conduct.html).

Please keep in mind that lesson Maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum. If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.

</details>

### CODE REVIEW

**Comment by themkots** [comment]
As part of the Instructor Checkout process, I suppose this can be seen as a contribution.... Ta.

File: episodes/04-pipefilter.md [hunk 1]
```diff
@@ -6,16 +6,17 @@ exercises: 10
 
 ::::::::::::::::::::::::::::::::::::::: objectives
 
+- Explain the advantage of linking commands with pipes and filters.
+- Combine commands in more than one stages to get new output
```
Comment (bkmgit) [inline episodes/04-pipefilter.md:10]: Perhaps use
```suggestion
- Combine sequences of commands to get new output
```

File: episodes/04-pipefilter.md [hunk 2]
```diff
@@ -6,16 +6,17 @@ exercises: 10
...
-- Construct command pipelines with two or more stages.
 - Explain what usually happens if a program or pipeline isn't given any input to process.
-- Explain the advantage of linking commands with pipes and filters.
 
 ::::::::::::::::::::::::::::::::::::::::::::::::::
 
 :::::::::::::::::::::::::::::::::::::::: questions
 
-- How can I combine existing commands to do new things?
+- How can I produce various forms of the output of existing commands? How to show only part of the output? ( -> command line switches and pipes)
```
Comment (bkmgit) [inline episodes/04-pipefilter.md:18]: Perhaps use:
```suggestion
- How can I combine existing commands to produce a desired output?
- How can I show only part of the output? 
```

File: episodes/04-pipefilter.md [hunk 3]
```diff
@@ -6,16 +6,18 @@ exercises: 10
...
-- Explain the advantage of linking commands with pipes and filters.
 
 ::::::::::::::::::::::::::::::::::::::::::::::::::
 
 :::::::::::::::::::::::::::::::::::::::: questions
 
-- How can I combine existing commands to do new things?
+- How can I combine existing commands to produce a desired output?
+- How can I show only part of the output? 
+- How can I combine existing commands to do new things? ( -> pipes)
```
Comment (bkmgit) [inline episodes/04-pipefilter.md:20]: This seems redundant, can possibly be removed.
```suggestion

```

**Comment by bkmgit** [comment]
@themkots  Thanks for the contribution.  Feel free to make an additional pull request if last change can be improved upon.

**Pull request merged**